keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/SliceVertex.java

```diff
@@ -81,4 +86,13 @@ private DoubleTensor padSliceWithZerosToMatchOriginalShape(DoubleTensor tensor)
         return outputTensor;
     }
 
+    @SaveVertexParam(DIMENSION_NAME)
+    public int getDimension() {
+        return dimension;
```

Comment:
This is a general comment, and it's coming a little late, but - would it be possible to annotate the field `this.dimension` so you don't need to add public methods? And/or can the method be private?

Reply:
As discussed - this was due to a concern about ripping out private params and having getters in general is pretty "Javonic" :)